src/v/serde/rw/vector.h

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "serde/rw/reservable.h"
+#include "serde/rw/rw.h"
+#include "serde/serde_exception.h"
+#include "serde/serde_size_t.h"
+#include "serde/type_str.h"
+#include "ssx/sformat.h"
+#include "vlog.h"
+
+#include <cinttypes>
+#include <limits>
+#include <type_traits>
+
+namespace serde {
+
+template<typename T>
+concept Vector = requires(T t) {
```

Comment:
nit: consider adding a comment this is more than just `std::vector` like the commit message says - it's any vector like container like `ss::chunked_fifo`, `fragmented_vector`, `ss::circular_buffer`, etc

Reply:
Added a comment, thank you!